src/st/TranslateLinks.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WPML\PB;
+
+class TranslateLinks {
+
+	/**
+	 * @param \WPML_ST_String_Factory $stringFactory
+	 * @param array $activeLanguages
+	 *
+	 * @return \Closure
+	 */
+	public static function getLinkTranslation( \WPML_ST_String_Factory $stringFactory, $activeLanguages ) {
```

Comment:
@BruceOnTheGo I find this approach a bit hard to understand. Not really because of the code, but because of the function name.

By using a function called `getLinkTranslation`, I would expect to pass a link (as a string) as the input and get a translation as the output.

Instead, I must pass an instance of `WPML_ST_String_Factory` together with an array of active languages and then I get back another function.

Then I must call this new function with a string ID (not a link, nor a string: an ID) and then I'm completely lost, because I see no more return points.

Reply:
@andreasciamanna How about `getLinkTranslator` because it returns a function to translate links in strings.

Reply:
Or `getTranslatorForString`

`TranslateLinks::getTranslatorForString()`

Reply:
@BruceOnTheGo the "-or" suffix may help.

Reply:
@andreasciamanna Done